### PR TITLE
ML skill fix

### PR DESCRIPTION
**Pull request by 5A11**
This PR fixes a problem with the ML skill #2508. 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/skills/ml_train/handlers.py [hunk 1]
```diff
@@ -129,36 +129,25 @@ def _handle_terms(
...
-            )
-            terms_ = strategy.terms_from_proposal(ml_trade_msg.terms)
-            ml_trade_dialogue.terms = terms_
-            _, ledger_api_dialogue = ledger_api_dialogues.create(
-                counterparty=LEDGER_API_ADDRESS,
-                performative=LedgerApiMessage.Performative.GET_RAW_TRANSACTION,
-                terms=terms_,
-            )
-            ledger_api_dialogue = cast(LedgerApiDialogue, ledger_api_dialogue)
-            ledger_api_dialogue.associated_ml_trade_dialogue = ml_trade_dialogue
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/ml_train/handlers.py:161]: What happened to that? Why can it be removed?

File: packages/fetchai/skills/ml_train/behaviours.py [hunk 2]
```diff
@@ -64,8 +64,7 @@ def act(self) -> None:
             if not result.successful():
                 return
 
-            ml_task = result.get()
-            strategy.weights = ml_task.result
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/ml_train/behaviours.py:68]: So the property result does not need to be used?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: packages/fetchai/skills/ml_train/behaviours.py (hunk 2 above)
Comment (5A11) [reply packages/fetchai/skills/ml_train/behaviours.py:68]: Yep

File: packages/fetchai/skills/ml_train/handlers.py (hunk 1 above)
Comment (5A11) [reply packages/fetchai/skills/ml_train/handlers.py:161]: Not sure why this block was there. 

This logic (sending GET_RAW_TRANSACTION) is what happens when we add the ml_dialogue to the tx_behaviour for processing (two lines below).

Comment (DavidMinarsch) [reply packages/fetchai/skills/ml_train/handlers.py:161]: Well spotted!

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**